Shared/ModuleHandle.ps1

```diff
@@ -0,0 +1,112 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+This script defines a function `Request-Module` that checks for the presence of specified PowerShell modules.
+If a module is not found, it attempts to install it on current user scope.
+The function accepts a list of module names and an optional minimum version for the modules.
+It returns a boolean indicating whether all specified modules are installed successfully.
+
+.PARAMETER Modules
+ Mandatory array of strings specifying the names of the modules to check and install if necessary.
+.PARAMETER MinModuleVersion
+ Optional parameter to specify the minimum version of the modules (default is null).
+
+.OUTPUTS
+bool. A boolean indicating whether all specified modules are installed successfully.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement"
+This example checks if the "ExchangeOnlineManagement" module is installed. If it is not found, the script attempts to install it.
+
+.EXAMPLE
+$requestModule = Request-Module -Modules "ExchangeOnlineManagement" -MinModuleVersion $MinModuleVersion
+This example checks if the "ExchangeOnlineManagement" module with a specified minimum version is installed. If it is not found, the script attempts to install it.
+#>
+
+. $PSScriptRoot\..\Shared\Confirm-Administrator.ps1
+
+function Request-Module {
+    [OutputType([bool])]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string[]]$Module,
+        [Parameter(Mandatory = $false)]
+        [System.Version]$MinModuleVersion = $null,
+        [Parameter(Mandatory = $false)]
+        [switch]$InstallAllUsersIfNotAvailable
+    )
+
+    $noFoundError = $true
+
+    foreach ($m in $Module) {
```

Comment:
Sorry to be so petty. Wouldn't it be better to return a final statement for each module that was passed when calling the `Request-Module` function? We support processing multiple modules in here but return just one state. If one of the modules fail and the remaining ones are processed successfully, we just return `$false` because one of them has failed. I think it would be better to put the module and final state to a hashtable and finally return it. What do you think?

Reply:
Hi @lusassl-msft do not worry, it is a healthy discussion looking for the best solution, no problem at all 😉
Maybe it is easier to simplify and just allow one module each time instead of an array and return true or false for each one.
I think we are not going to request several modules.
What do you think?

Reply:
That could make it more complicated for the caller to use. If that is the case, I would just adjust to only accept a single module and you have to test against each one.

Reply:
Yep, let’s move to the one module approach then 👍

Reply:
@iserrano76 are you able to make this change?

Reply:
Yes, I am trying to find some time for testing.